src/worker.cpp

```diff
@@ -239,37 +239,29 @@ std::string Worker::getHWVersion(const types::IPZVpdMap& parsedVpd) const
     return hwString.str();
 }
 
-void Worker::getVpdDataInVector(const std::string& vpdFilePath,
-                                types::BinaryVector& vpdVector,
-                                size_t& vpdStartOffset)
+void Worker::fillVPDMap(const std::string& vpdFilePath,
+                        types::VPDMapVariant& vpdMap)
 {
-    std::fstream vpdFileStream;
-    vpdFileStream.exceptions(std::ifstream::badbit | std::ifstream::failbit);
+    logging::logMessage(std::string("Parsing file = ") + vpdFilePath);
 
-    try
+    if (vpdFilePath.empty())
     {
-        vpdFileStream.open(vpdFilePath,
-                           std::ios::in | std::ios::out | std::ios::binary);
-        auto vpdSizeToRead = std::min(std::filesystem::file_size(vpdFilePath),
-                                      static_cast<uintmax_t>(65504));
-        vpdVector.resize(vpdSizeToRead);
-
-        vpdFileStream.seekg(vpdStartOffset, std::ios_base::beg);
-        vpdFileStream.read(reinterpret_cast<char*>(&vpdVector[0]),
-                           vpdSizeToRead);
-
-        vpdVector.resize(vpdFileStream.gcount());
-        vpdFileStream.clear(std::ios_base::eofbit);
+        throw std::runtime_error("Invalid file path passed to fillVPDMap API.");
```

Comment:
do we need to mention API name in logging message?

Reply:
right, but you have the api name logged at line 249. you need to remove it

Reply:
Ohh, that... Does that actually matters? Even if we have or we don't have?

Reply:
if the log is internal to our team, then it's fine, but this is an open log which doesn't add any usage, rather giving file path in the same line is useful. I leave it up to you